Src/Microsoft.Dynamic/Actions/Calls/ActualArguments.cs

```diff
@@ -100,33 +101,38 @@ public sealed class ActualArguments {
                 if (paramIndex >= 0) {
                     int nameIndex = paramIndex - positionalArgCount;
 
-                    // argument maps to already bound parameter:
-                    if (paramIndex < positionalArgCount || boundParameters[nameIndex]) {
-                        if (duppedNames == null) {
-                            duppedNames = new List<string>();
-                        }
+                    if (paramIndex < positionalArgCount) {
+                        // argument maps to already bound positional parameter
+                        duppedNames ??= new List<string>();
+                        duppedPositionals ??= new List<int>(duppedNames.Count);
                         duppedNames.Add(ArgNames[i]);
+                        duppedPositionals.Add(method.PositionOfParameter(ArgNames[i]));
+                    } else if (boundParameters[nameIndex]) {
+                        // argument maps to already bound named parameter
+                        duppedNames ??= new List<string>();
+                        duppedNames.Add(ArgNames[i]);
+                        duppedPositionals?.Add(0);
```

Comment:
Is it possible for the two lists to go out of sync? For example if this branch is taken first and then the other branch? If not then wouldn't the following hold true?
```C#
Debug.Assert(duppedPositionals is null || duppedNames.Length == duppedPositionals.Length);
```

Reply:
My mistake. I forgot that the number given to the List constructor is the capacity, not size. The assertion you wrote was supposed to hold. Although strictly speaking `duppedPositionals` are allowed to be shorter than `duppedNames`, this is not what the code here intends to do.

Thanks for checking!